### PR TITLE
Implement SimpleDatasetReader

### PR DESCRIPTION
**Pull request by ruimashita**
which is read batch size data from Dataset object simply (not using thread and multiprocess) when DatasetIterator is not enable_prefetch.

## Motivation and Context

As in #122, we implement DatasetIterator class, But when we use enable_prefetch is False, it is not good for debugging because dataset iterator using thread.

## Description

* Implement `_SimpleDatasetReader` 
* Implement unit tests.
  * `test_dataset_iterator_batch_size()` which is asset that dataset iterator give batch size data.
  * `test_dataset_iterator_batch_orde()` which is asset that data given by iterator is same whether enabele_prefetch ture or false.
* Delete `_SingleProcessDatasetPrefetchThread` class, it is not used.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ruimashita** [comment]
@iizukak @nlpng @lm-jira Plz review 👍

**Comment by iizukak** [comment]
@ruimashita 
`_SimpleDatasetReader_` 's name is maybe too generic.
Is there more descriptive name for this class?  I have no idea...

**Comment by ruimashita** [comment]
@iizukak
Yes 👍 the class has bad name, I thought about class name a lot, But I didn't come up with.
ok, I will re-think

File: lmnet/lmnet/datasets/dataset_iterator.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 from multiprocessing import Pool
```
Comment (nlpng) [inline lmnet/lmnet/datasets/dataset_iterator.py]: Should this have the copyright label as well?

Comment (ruimashita) [reply lmnet/lmnet/datasets/dataset_iterator.py]: @nlpng 
Oh, thanks. I will add copyright.

Comment (ruimashita) [reply lmnet/lmnet/datasets/dataset_iterator.py]: changed https://github.com/blue-oil/blueoil/pull/166/commits/667d13857325ceccfa463d49024bcfb9d0556ffe

**Pull request merged**